googetdb/googetdb.go

```diff
@@ -107,7 +109,9 @@ func (g *GooDB) addPkg(pkgState client.PackageState) error {
 	spec := pkgState.PackageSpec
 
 	pkgState.InstalledApp.Name, pkgState.InstalledApp.Reg = system.AppAssociation(spec, pkgState.LocalPath)
-	pkgState.InstallDate = time.Now().Unix()
+	if pkgState.InstallDate == 0 {
+		pkgState.InstallDate = time.Now().Unix()
```

Comment:
InstallDate is a new field that old state files shouldn't have. I'm not sure if we'd ever run into a case where this code is used currently, but I guess this is okay for future use.

Reply:
The other purpose of this check is to prevent WriteStateToDB from overwriting the old InstallDate with the current time whenever we read in the installed packages state with db.FetchPkgs and then write it back out with db.WriteStateToDB.